iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs

```diff
@@ -168,15 +162,10 @@ public static void UpdateMessageHeaderAndProperties(AmqpMessage amqpMessage, Mes
         /// <summary>
         /// Copies the Message instance's properties to the AmqpMessage instance.
         /// </summary>
-        public static void UpdateAmqpMessageHeadersAndProperties(AmqpMessage amqpMessage, Message data, bool copyUserProperties = true)
+        public static void UpdateAmqpMessageHeadersAndProperties(AmqpMessage amqpMessage, OutgoingMessage data, bool copyUserProperties = true)
         {
             amqpMessage.Properties.MessageId = data.MessageId;
 
-            if (data.To != null)
-            {
-                amqpMessage.Properties.To = data.To;
```

Comment:
only relevant for C2D message: https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct#system-properties-of-d2c-iot-hub-messages